open-craft/opencraft
# [BB-3868] [NewConsole] Move Button Customization component to sidebar

This PR implements the `ButtonCustomizationPage` for new console and moves the color inputs to sidebar.

**JIRA Tickets** [BB-3868](https://tasks.opencraft.com/browse/BB-3868) [Gitlab#720](https://gitlab.com/opencraft/dev/opencraft/-/issues/720)

**Testing Instructions**
1. Checkout to this branch
2. Start the OCIM backend and frontend servers.
3. Make sure that a test instance is created in OCIM admin.
4. Visit http://localhost:3000/newconsole/theming/buttons/edit
5. Make sure that all button customization components are present in sidebar.
6. Verify that changing `ColorInputField` values changes the theme data in `BetaTestApplication`

**Author Notes**
1. While implementing the changes, I noticed that some styles for `ColorInputField` component were added in all the components where `ColorInputField` was imported and used. The styling of this field was also causing issues when moving to the sidebar.
2. This PR also contains changes to fix `ColorInputField` such that it doesn't require additional styling in importing component.
3. I found the old `ButtonStyles` component design didn't look good in sidebar. Thus, I added a new `ButtonStyles` component in `newConsole` with comparatively better design.
4. This PR is dependent on #719 for live preview in static pages.

**Screenshots**

ButtonsStyle Component collapsed in Sidebar
![buttons-collapsed](https://user-images.githubusercontent.com/18226212/111920364-88df1600-8ab4-11eb-8462-0fb543a61116.png)

ButtonStyle Component expanded in Sidebar
![Buttons-Expanded](https://user-images.githubusercontent.com/18226212/111920456-13277a00-8ab5-11eb-8431-c30f98184f0c.png)
